### ISSUE TITLE
Auto-correction of Layout/ClassStructure cop breaks codes

### ISSUE THREAD

**Issue by yujideveloper** [rubocop/rubocop#8034]
## Steps to reproduce the problem

I ran `rubocop -a --only Layout/ClassStrucutre` on the following codes.

``` ruby
# example.rb
class Foo
  C1 = "abc".freeze
  C2 = "efg".freeze

  def self.class_m1
    puts __callee__
  end

  def self.class_m2
    puts __callee__
  end

  def instance_m1
    puts __callee__
  end

  private

  def private_m1
    puts __callee__
  end

  def initialize(args)
    @args = args
  end

  def private_m2
    puts __callee__
  end
end
```

## Expected behavior

I expected no error and expected get following codes.

``` ruby
class Foo
  C1 = "abc".freeze
  C2 = "efg".freeze

  def initialize(args)
    @args = args
  end

  def instance_m1
    puts __callee__
  end

  private

  def private_m1
    puts __callee__
  end

  def private_m2
    puts __callee__
  end

  def self.class_m1
    puts __callee__
  end

  def self.class_m2
    puts __callee__
  end
end
```

## Actual behavior

I encountered the following results and broken codes.

```
$ rubocop -a --only Layout/ClassStructure example.rb
Inspecting 1 file
E

Offenses:

example.rb:8:3: C: [Corrected] Layout/ClassStructure: initializer is supposed to appear before private_methods.
  def initialize(args) ...
  ^^^^^^^^^^^^^^^^^^^^
example.rb:9:3: C: [Corrected] Layout/ClassStructure: initializer is supposed to appear before private_methods.
  def initialize(args) ...
  ^^^^^^^^^^^^^^^^^^^^
example.rb:16:3: C: [Corrected] Layout/ClassStructure: constants is supposed to appear before private_methods.
  C1 = "abc".freeze
  ^^^^^^^^^^^^^^^^^
example.rb:17:3: C: [Corrected] Layout/ClassStructure: constants is supposed to appear before private_methods.
  C1 = "abc".freeze
  ^^^^^^^^^^^^^^^^^
example.rb:23:3: C: [Corrected] Layout/ClassStructure: initializer is supposed to appear before private_methods.
  def initialize(args) ...
  ^^^^^^^^^^^^^^^^^^^^
example.rb:33:1: E: Lint/Syntax: unexpected token $end
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)

1 file inspected, 6 offenses detected, 5 offenses corrected
```

``` ruby
class Foo
  private

  def private_m1
    puts __callee__
  end

  def initialize(args)
    @args = args
  end

  def private_m2
    puts __callee__
  end

  C1 = "abc".freeze
  C2 = "efg".freeze

  def self.class_m1
    puts __callee__
  end

  def self.class_m2
    puts __callee__
  end

  def instance_m1
    puts __callee__
  end

  C1 = "abc".freeze
  C1 = "abc".freezeend
```

## RuboCop version
```
$ [bundle exec] rubocop -V
0.84.0 (using Parser 2.7.1.2, rubocop-ast 0.0.3, running on ruby 2.5.7 x86_64-linux)
```

**Comment by yujideveloper** [comment]
Maybe this is related to #6962.
